go.mod

```diff
@@ -3,6 +3,7 @@ module sigs.k8s.io/container-object-storage-interface-provisioner-sidecar
 go 1.18
 
 require (
+	github.com/agiledragon/gomonkey/v2 v2.12.0
```

Comment:
I have reservations about using a monkey patch library. I realize that monkey patching is a common trend in python and ruby development (and other similar scripting languages), but my understanding is that the Golang devs specifically didn't allow for similar functionality. 

When I have investigated monkey patching for unit tests in Go, the advice I have seen is to instead use interfaces (even defining custom ones for unit tests if needed) and mocking using Go's builtin language tools.

Reply:
Sure, will incorporate this in my next commit.